feat: adds masking and auth for audit reports

as1729 opened the pull request:
### Ticket #1132
## Description
- Ensures that only logged-in users can receive a presigned S3 URL to the audit report

## Screenshots / Demo Video

## Testing

### Automated and Unit Tests
- [x] Added Unit tests

### Manual tests for Reviewer
- [ ] Added steps to test feature/functionality manually

## Checklist
- [x] Provided ticket and description
- [x] Provided screenshots/demo
- [ ] Provided testing information
- [x] Provided adequate test coverage for all new code
- [x] Added PR reviewers

as1729 commented:
@TylerHendrickson there's a little more work to be done to this PR around fixing tests etc. But the overall flow is working as intended so the code here should stay more or less the same with additional test directory changes.

TylerHendrickson commented on packages/server/src/arpa_reporter/routes/audit-report.js:
```diff
@@ -4,9 +4,42 @@ const express = require('express');
+        await s3.headObject(baseParams).promise();
+    } catch (error) {
+        console.log(error);
+        res.redirect(`${process.env.WEBSITE_DOMAIN}/arpa_reporter?alert_text=The audit report you requested does not exist. Please try again by clicking the 'Send Audit Report By Email'.&alert_level=err`);
```
Not sure if this is URL-encoded by default; if not, it may need to be.

TylerHendrickson replied:
Additionally, it may be worth replacing "does not exist" with "has expired". It's a small change, but I think it communicates the situation a bit more directly and possibly has a lower chance of being misinterpreted.

as1729 replied:
Done 👍

TylerHendrickson reviewed (commented):
Two small thoughts, but nothing that I think would need to hold this up.

as1729 commented:
@TylerHendrickson  The PR should be ready to merge. I tested the main functionality locally and added the following loom video. I also added unit tests to cover most of the new functions added.

https://www.loom.com/share/622880c4c8bc41c5a943bab3391ef17c

TylerHendrickson reviewed (approved):
Looks good!
